browser/search_engines/search_engine_tracker.cc

```diff
@@ -44,6 +44,40 @@ void RecordSearchEngineP3A(const GURL& search_engine_url,
   UMA_HISTOGRAM_ENUMERATION(kDefaultSearchEngineMetric, answer);
 }
 
+void RecordSwitchP3A(const GURL& to_url, const GURL& from_url) {
```

Comment:
would be great to try to reenable the existing tests and add checks for this new metric

Reply:
Indeed. `SearchEngineProviderP3ATest` seems to run fine locally. Any ideas why [it was unreliable](https://github.com/brave/brave-browser/issues/13057)? I assume it was racing with the template service initialisation? 

Ideas:
- Re-enable and add logging to see if we're failing to get a url to record at `SearchEngineTracker` construction time.
- inject some dummy content and spin the event loop, hoping that's enough to settle the ordering.
- set some testing-only flag when a measurement is recorded and wait for it in the test (yuck!)

Reply:
Definitely worth debugging! I don't think anybody every dug in / found a root cause

Reply:
most probably we just need `search_test_utils::WaitForTemplateURLServiceToLoad();`

Reply:
Nice! I've submitted that as #10328. Will add a test for this and push once I have the default measurement added.

Reply:
This has now landed. Hopefully it sticks!